DQMOffline/Trigger/plugins/BPHMonitor.cc

```diff
@@ -146,55 +314,334 @@ void BPHMonitor::bookHistograms(DQMStore::IBooker     & ibooker,
 #include "FWCore/Framework/interface/EventSetup.h"
 #include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
 #include "Geometry/Records/interface/TrackerTopologyRcd.h"
+#include "TLorentzVector.h"
 void BPHMonitor::analyze(edm::Event const& iEvent, edm::EventSetup const& iSetup)  {
 
-//  edm::Handle<reco::BeamSpot> const& beamSpot;
-  // Filter out events if Trigger Filtering is requested
-  if (den_genTriggerEventFlag_->on() && ! den_genTriggerEventFlag_->accept( iEvent, iSetup) ) return;
-
-  
   edm::Handle<reco::BeamSpot> beamSpot;
   iEvent.getByToken( bsToken_,  beamSpot);
   edm::Handle<reco::MuonCollection> muoHandle;
   iEvent.getByToken( muoToken_, muoHandle );
+    
 
-  for (auto const & m : *muoHandle) {
-    if(!muoSelection_ref(m))continue;
-    muPhi_.denominator->Fill(m.phi());
-    muEta_.denominator->Fill(m.eta());
-    muPt_.denominator ->Fill(m.pt());
-    const reco::Track * track = 0;
-    if (m.isTrackerMuon()) track = & * m.innerTrack();
-    else if (m.isStandAloneMuon()) track = & * m.outerTrack();
-    if (track) {
-      double d0 = track->dxy(beamSpot->position());
-      double z0 = track->dz(beamSpot->position());
-      mud0_.denominator ->Fill(d0);
-      muz0_.denominator ->Fill(z0);
-    }
-  } 
-  // applying selection for numerator
+//  edm::Handle<reco::BeamSpot> const& beamSpot;
+  // Filter out events if Trigger Filtering is requested
+  if (tnp_>0) {
   if (num_genTriggerEventFlag_->on() && ! num_genTriggerEventFlag_->accept( iEvent, iSetup) ) return;
-  for (auto const & m : *muoHandle) {
-    if(!muoSelection_(m))continue;
-    muPhi_.numerator->Fill(m.phi());
-    muEta_.numerator->Fill(m.eta());
-    muPt_.numerator ->Fill(m.pt());
-    const reco::Track * track = 0;
-    if (m.isTrackerMuon()) track = & * m.innerTrack();
-    else if (m.isStandAloneMuon()) track = & * m.outerTrack();
-    if (track) {
-      double d0 = track->dxy(beamSpot->position());
-      double z0 = track->dz(beamSpot->position());
-      mud0_.numerator ->Fill(d0);
-      muz0_.numerator ->Fill(z0);
+  std::vector<reco::Muon> tagMuons;
+  std::vector<reco::Muon> probeMuons;
+  for ( auto const & m : *muoHandle ) {
+  if ( muoSelection_tag( m ) ) tagMuons.push_back(m);
+  }
+  for (int i = 0; i<int(tagMuons.size());i++){
+    for ( auto const & m : *muoHandle ) { 
+      if ((tagMuons[i].pt() - m.pt())<=0.01)continue;//not the same  
```

Comment:
can we have this "magic" numbers set as global variables
and not having them hard coded, please ?